src/surface.rs

```diff
@@ -165,6 +171,34 @@ impl Surface {
         unsafe { ffi::cairo_surface_get_fallback_resolution(self.to_raw_none(), &mut x_pixels_per_inch, &mut y_pixels_per_inch); }
         (x_pixels_per_inch, y_pixels_per_inch)
     }
+
+    pub fn create_similar_image(&self, format: Format, width: i32, height: i32) -> Option<Surface> {
+        unsafe {
+            let p = ffi::cairo_surface_create_similar_image(self.to_raw_none(), format.into(), width, height);
+            if p.is_null() {
+                None
+            } else {
+                Some(Self::from_raw_full(p))
+            }
+        }
+    }
+
+    pub fn map_to_image(&self, extents: Option<RectangleInt>) -> Option<MappedImageSurface> {
+        unsafe {
+            let p = match extents {
+                Some(ref e) => ffi::cairo_surface_map_to_image(self.to_raw_none(), e.to_raw_none()),
+                None => ffi::cairo_surface_map_to_image(self.to_raw_none(), 0 as *const _),
+            };
+            if p.is_null() {
```

Comment:
If you read the docs: this function never returns `NULL` but like with many other cairo functions you have to check the status of the returned surface.

Reply:
I.e. it would also return a `Result` and not an `Option`